presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -215,6 +241,16 @@ private Table getTable(SemiTransactionalHiveMetastore metastore, SchemaTableName
         return table;
     }
 
+    private static List<String> getPartitionNamesByParts(SemiTransactionalHiveMetastore metastore, SchemaTableName tableName, List<String> partitionValue)
```

Comment:
As i understand it implicitly adds support for wildcards in the partition values. Do we want this? Just want to make sure that we are making this decision consciously.

Reply:
@findepi I forgot to mention in the commit message. The default behavior of `ANALYZE foo` without any analyze properties is to analyze the whole table. I can imagine partition ranges to be useful but since the property is typed, it's hard to define what type should the property be (expression? varchar?).

Reply:
I didn't realize that will implicitly support wildcard. I didn't see the logic but I guess it's in the Glue client. 

I use metastore to resolve the partition name because I want to know if the partition specified by user exists or not. I can directly call `makePartName()` to construct a partition name. Will that avoid the implicit wildcard? The downside is that we will only detect non-existing partition when getting splits. And the error message will be `Partition no longer exists...`, which is a little bit confusing.

Reply:
> The default behavior of ANALYZE foo without any analyze properties is to analyze the whole table. 

Thanks!

> I can imagine partition ranges to be useful but ..

i am fine if we don't support ranges. We can revisit later.

>  use metastore to resolve the partition name because ..

Just a side question -- how do i specify partition that has null value for the partitioning key? Is it via this `'__DEFAULT_PARTITION__'`? Make sure you have a test covering this.

Reply:
Thanks for reminding! The partition name is `'__HIVE_DEFAULT_PARTITION__'` for `null` partitions and I added a test case for it. 

However, this make me think if user should specify a partition by providing a list of partition column values or just the partition name. Using the partition name directly seems more straight forward. Although it's not very user friendly.

Reply:
Did you push this test to the PR already? How does user specify the null partition key (using `array[null]` or `array['__HIVE_DEFAULT_PARTITION__']`, or..)?

Reply:
@findepi . It's `array['__HIVE_DEFAULT_PARTITION__']`. Right now I am debugging issues with my production tests. Will update the PR soon.

Reply:
would you consider taking `array[null]` for the null part key? Normally, "__HIVE_DEFAULT_PARTITION__" is a token that user should not need to know that it exists. 
cc @electrum